Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM fedora:30
+
+RUN dnf -y update && \
+    dnf -y install make go
+
+COPY . ./app
+
+WORKDIR ./app
+
```

Comment:
please implement go testing before starting the application

Reply:
Testing the application in the Dockefile isn't the proper way to do this.

Generally, the developer would:
1. Write new functionality
2. Run `go test` as they develop the new unit tests to make sure everything passes
3. Use docker as the deployment method after everything is working as expected

Otherwise, a second `Dockerfile.dev` can be made to do all of this if you want the development to be fully containerized as well. This new Dockerfile would then spin up the application and do all the testing you want, so the developer doesn't need to install any local dependencies.

Reply:
The reason why we want test just before the run command in docker is just to ensure the user that the build he is using for deployment has passed all the unit-tests in order to ensure that no preliminary errors exist in the deployment build. 

Note: we dont aim for testing in docker. This is just to ensure that the latest version he is building in docker is free from all basic probs which have been covered in the unit-testing.